assets/js/components/Modal.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Modal component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { createPortal, useEffect, useState } from '@wordpress/element';
+
+function Modal( { children } ) {
+	// Using state as we need `el` to not change when the component re-renders
+	const [ el ] = useState( document.createElement( 'div' ) );
+
+	useEffect( () => {
+		const root = document.querySelector( '.googlesitekit-plugin' ) || document.body;
+		root.appendChild( el );
+
+		return () => root.removeChild( el );
+	}, [] );
```

Comment:
This looks good as far as I can tell. @tofumatt Would be great to get your review here as well.